docs/src/modules/components/DemoEditor.tsx

```diff
@@ -32,6 +32,7 @@ const StyledMarkdownElement = styled(MarkdownElement)(({ theme }) => ({
 
 const StyledSimpleCodeEditor = styled(SimpleCodeEditor)(({ theme }) => ({
   ...theme.typography.body2,
+  letterSpacing: '0.01071em', // must sync with MarkdownElement.js#L42, otherwise the text will overlap when highlight the code.
```

Comment:
Why not do the opposite? Remove `letterSpacing: '0.01071em'` in MarkdownElement.js#L42? I don't understand why this style was added in #34976

Reply:
> and then make sure Material UI + branding theme is always the parent context.

Can we not do this? It is very hard to reason about and produce a different result when the component renders in a different context.

The reason I use lightTheme and darkTheme is to make the component agnostic of the context. This means the component can render in any place without the theme context and it still produces the same appearance.

Reply:
> Why was letterSpacing: '0.01071em', MarkdownElement.js#L42, added in https://github.com/mui/material-ui/pull/34976?

It is because `theme.typography.body2` in Material UI contains `letterSpacing: '0.01071em'` but the branding theme does not so I have to add it here so that we preserve the existing behavior and Argos regression pass.

Reply:
Let's break it. I am removing the letter-spacing.

Reply:
> Can we not do this? It is very hard to reason about and produce a different result when the component renders in a different context.

The meat of this proposal was to make the page throw when the theme in the context provided to a docs-infra component, e.g. MarkdownElement isn't the brandingTheme. For example, why are 1 and 2 having two different values:

1. https://master--material-ui.netlify.app/blog/making-customizable-components/ padding 20px

<img width="163" alt="Screenshot 2022-11-15 at 00 57 28" src="https://user-images.githubusercontent.com/3165635/201793109-a1ddddcc-88ec-4627-89a6-d57be1b93f2d.png">

2. https://master--material-ui.netlify.app/material-ui/react-checkbox/#accessibility padding 16px

<img width="137" alt="Screenshot 2022-11-15 at 00 57 40" src="https://user-images.githubusercontent.com/3165635/201793106-545ca904-868a-4012-95c3-f6f9f0079cd7.png">

I shared a possible implementation in https://github.com/mui/material-ui/pull/34976#issuecomment-1314519718. My concern is that importing the theme solves the problem but it feels like we don't solve the root problem, we branch away from the default API we recommend people use the component. I fear that it makes it harder to solve as it hides the instances where the value in the context is wrong.

The flip side is that we are getting closer to https://vanilla-extract.style/ with this theme import approach.

Reply:
> The meat of this proposal was to make the page throw when the theme in the context provided to a docs-infra component

Seems like we have different thoughts on this. Better to discuss it in a proper place to include other opinions and create documentation around it 👉 added to [Notion page](https://www.notion.so/mui-org/Site-styling-architecture-e781897a38904671ba909c51dcb28f83).

Reply:
@siriwatknp I have completed the notion page. How do you want to move from there?

Reply:
1. I will start documenting "How it currently works" 
2. Create options that we want to have in the future
3. Include X and Toolpad team to take a look, give feedback, and discuss until we agree on the final decision.
4. Start implementing (likely 1st-2nd quarter 2023). Keep in mind that whichever option we choose, might impact the Toolpad and X pages so there will be some migration period.